pkg/cloudprovider/aws/aws.go

```diff
@@ -97,6 +184,217 @@ func (self *GoamzEC2) GetMetaData(key string) ([]byte, error) {
+	}
+
+	request := &elb.DescribeLoadBalancer{}
+	// Names are limited to 32 characters, so we must use tags
+	//request.Names = []string{findName}
+	response, err := client.DescribeLoadBalancers(request)
+	if err != nil {
+		elbError, ok := err.(*elb.Error)
+		if ok && elbError.Code == "LoadBalancerNotFound" {
+			// Not found
+			return nil, nil
+		}
+		glog.Error("error describing load balancers: ", err)
+		return nil, err
+	}
+
+	loadBalancersByAwsId := map[string]elb.LoadBalancer{}
+	for _, loadBalancer := range response.LoadBalancers {
+		awsId := loadBalancer.LoadBalancerName
+		if !strings.HasPrefix(awsId, LOADBALANCER_NAME_PREFIX) {
+			continue
+		}
+
+		// TODO: Cache the name -> tag mapping (it should never change)
+		loadBalancersByAwsId[awsId] = loadBalancer
+	}
+
+	loadBalancersByName := map[string]elb.LoadBalancer{}
+	if len(loadBalancersByAwsId) != 0 {
+		describeTagsRequest := &elb.DescribeTags{}
+		describeTagsRequest.LoadBalancerNames = []string{}
+		for awsId := range loadBalancersByAwsId {
+			describeTagsRequest.LoadBalancerNames = append(describeTagsRequest.LoadBalancerNames, awsId)
+		}
+		describeTagsResponse, err := client.DescribeTags(describeTagsRequest)
+		if err != nil {
+			glog.Error("error describing tags for load balancers: ", err)
+			return nil, err
+		}
+
+		if describeTagsResponse.NextToken != "" {
+			// TODO: Implement this
+			err := fmt.Errorf("error describing tags for load balancers - pagination not implemented")
+			return nil, err
+		}
+
+		for _, loadBalancerTag := range describeTagsResponse.LoadBalancerTags {
+			awsId := loadBalancerTag.LoadBalancerName
+			name := ""
+			for _, tag := range loadBalancerTag.Tags {
+				if tag.Key == LOADBALANCER_TAG_NAME {
+					name = tag.Value
+				}
+			}
+			if name == "" {
+				glog.Warning("Ignoring load balancer with no k8s name tag: ", awsId)
+				continue
+			}
+
+			if findName != "" && name != findName {
+				continue
+			}
+
+			loadBalancer, ok := loadBalancersByAwsId[awsId]
+			if !ok {
+				// This might almost be panic-worthy!
+				glog.Error("unexpected internal error - did not find load balancer")
+				continue
+			}
+			loadBalancersByName[name] = loadBalancer
+		}
+	}
+	return loadBalancersByName, nil
+}
+
+// Implements EC2.CreateLoadBalancer
+func (self *goamzEC2) CreateLoadBalancer(region string, request *elb.CreateLoadBalancer) (string, error) {
+	client, err := self.getElbClient(region)
+	if err != nil {
+		return "", err
+	}
+
+	response, err := client.CreateLoadBalancer(request)
+	if err != nil {
+		glog.Error("error creating load balancer: ", err)
+		return "", err
+	}
+	return response.DNSName, nil
+}
+
+// Implements EC2.DeleteLoadBalancer
+func (self *goamzEC2) DeleteLoadBalancer(region string, name string) error {
+	client, err := self.getElbClient(region)
+	if err != nil {
+		return err
+	}
+
+	request := &elb.DeleteLoadBalancer{}
+	request.LoadBalancerName = name
+
+	_, err = client.DeleteLoadBalancer(request)
+	if err != nil {
```

Comment:
It shouldn't happen, so I added a TODO for now.  I'm worried it'll be almost impossible to hit in practice, so errors could lurk undetected in there.  Right now, it should be incredibly unlikely, and it should log a nice self-evident error message with the information we need to improve it!